infer/README.md

```diff
@@ -160,6 +160,8 @@ func (f *FileState) Annotate(a infer.Annotator) {
 }
 ```
 
+To deprecate a resource or field, annotate it with `Deprecate`.
```

Comment:
I think we should provide an example if we mention `Deprecate` here.

````suggestion
To deprecate a resource or field, annotate it with `Deprecate`:

```go
func (f *FileState) Annotate(a infer.Annotator) {
	a.Deprecate(&f.OldField, "You should prefer to use NewField instead.")
```
````

That said, I don't think that `infer/README.md` needs to document every feature. As long as it's well documented in the reference docs ([here](https://pkg.go.dev/github.com/pulumi/pulumi-go-provider@v0.25.0/infer#Annotator)), we should be ok.